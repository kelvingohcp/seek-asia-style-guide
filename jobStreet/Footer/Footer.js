import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';
import FooterLinks from './components/FooterLinks/FooterLinks';
import { getLocalization } from '../localization';
import _get from 'lodash/get';

const Footer = ({ language, country }) => {
  const year = new Date().getFullYear();
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <footer className={styles.container}>
        <FooterLinks footerMessages={footerMessages} />
        <p className={styles.copyright}>
          {_get(footerMessages, 'copyright').replace('{year}', year)}
        </p>
      </footer>
    );
  }

  return null;
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
