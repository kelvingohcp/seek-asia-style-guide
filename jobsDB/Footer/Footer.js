import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';

import FooterLinks from './components/FooterLinks/FooterLinks';
import FooterHK from './FooterHK/FooterHK';
import localization from './localization';

const Footer = ({ language, country, ...restProps }) => {
  const year = new Date().getFullYear();
  const messages = localization[`${language}-${country}`];

  if (country === 'hk') {
    return (<FooterHK language={language} country={country} {...restProps} />);
  }
  return (
    <footer className={styles.container}>
      <div className={styles.wrapper}>
        <div className={styles.pullRight}>
          <FooterLinks messages={messages} />
        </div>
        <div className={styles.pullLeft}>
          <p className={styles.copyright}>
            {messages['footer.copyright'].replace('{year}', year)}
            <span className={styles.fullversion}>
              {messages['footer.rightReserved']}
            </span>
          </p>
        </div>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
