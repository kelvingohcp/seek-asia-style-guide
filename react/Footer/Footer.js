import React from 'react';
import PropTypes from 'prop-types';
import GlobalFooter from '../private/GlobalFooter/GlobalFooter';
import { getLocalization } from '../../jobsDB/localization';
import { getLocalList } from '../../jobsDB/helper';
import _get from 'lodash/get';

const Footer = ({
  language,
  country,
  isExpandedVersion,
  showHeaderActionTrayOffset,
  showCountryLanguage,
  domainUrl,
  ...restProps }) => {
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <GlobalFooter
        country={country}
        isExpandedVersion={isExpandedVersion}
        showHeaderActionTrayOffset={showHeaderActionTrayOffset}
        showCountryLanguage={showCountryLanguage}
        footerMessages={footerMessages}
        locales={getLocalList({ messages, domainUrl, country, language })}
        {...restProps}
      />
    );
  }
  return null;
};

Footer.defaultProps = {
  country: 'hk',
  language: 'en',
  domainUrl: 'jobsdb.com',
  hasCompanyProfile: true,
  showHeaderActionTrayOffset: true,
  isExpandedVersion: false,
  showCountryLanguage: true
};

Footer.propTypes = {
  country: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  domainUrl: PropTypes.string.isRequired,
  hasCompanyProfile: PropTypes.bool,
  showHeaderActionTrayOffset: PropTypes.bool,
  isExpandedVersion: PropTypes.bool,
  showCountryLanguage: PropTypes.bool
};

export default Footer;
