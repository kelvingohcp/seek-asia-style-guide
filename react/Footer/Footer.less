@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "~seek-asia-style-guide/react/Header/components/ActionTray/ActionTray.less";
@wrapper-up-padding-top: @grid-base * 8;
@wrapper-up-padding-bottom: @grid-base * 3;

.container {
  cursor: default;
  margin: 0 auto;
  background-color: @saGrey6;
  width: 100%;
  font-weight: @sa-semi-bold;
}

.headerActionTrayOffset {
  @media @mobile {
    margin-bottom: @actionTrayHeight;
  }
}

.link {
  text-decoration: none;
}

.upperWrapper {
  padding-top: @wrapper-up-padding-top;
  padding-bottom: @wrapper-up-padding-bottom;

  -webkit-transition: all 0.3s ease-in-out;
  -moz-transition: all 0.3s ease-in-out;
  -ms-transition: all 0.3s ease-in-out;
  -o-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;

  &.hidden {
    max-height: 0;
    padding-top: 0;
    padding-bottom: 0;
    overflow: hidden;
    opacity: 0;
  }
}

.lowerWrapper {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
}

.lowerWrapperLinks {
  display: flex;
  flex-direction: column;
  line-height: @grid-base * 4;
  padding-left: @grid-base * 2;
  width: 100%;
  @media @tablet {
    flex-direction: row;
    line-height: normal;
    padding-bottom: @grid-base;
    width: 100%;
  }
  @media @desktop {
    padding-bottom: @grid-base;
    width: auto;
  }
}

.fullWidth {
  width: 100%;
}

.collapsed {
  padding-left: @grid-base * 4;
  @media @desktop {
    padding-left: @grid-base * 6;
  }
}

.chevronSvg {
  height: @grid-base * 3;
  width: @grid-base * 3;
}

.chevronIcon {
  cursor: pointer;
  padding-right: @grid-base * 4;
  padding-left: @grid-base * 8;
  display: flex;
  align-items: center;
  color: @saBlue2;
  @media @desktop {
    padding-right: @grid-base * 6;
    padding-left: @grid-base * 12;
  }
}

.endOfLine {
  display: flex;
  width: 100%;
  @media @tablet {
    width: 100%;
  }
  @media @desktop {
    width: auto;
  }
}

.endOfLineWithoutCountry {
  display: flex;
  width: 100%;
}

.localeIcon {
  margin: @grid-base @grid-base 0 0;
}

.currentLocale {
  display: none;
  @media @desktop {
    display: flex;
  }
}
