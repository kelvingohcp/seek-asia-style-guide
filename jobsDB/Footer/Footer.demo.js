import Footer from './Footer';
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as sketch from './Footer.sketch';

export const makeDummyLinkRendererForPath = path => {
  const DummyLinkRenderer = ({ href, children, ...props }) => (
    <Link style={{ textDecoration: 'none' }} to={`${path}#actualHref=${href}`} {...props}>
      {children}
    </Link>
  );

  DummyLinkRenderer.propTypes = {
    children: PropTypes.array,
    href: PropTypes.string
  };

  return DummyLinkRenderer;
};

const ROUTE = '/jobsdb-footer';

export default {
  route: ROUTE,
  tenantPath: 'jobsDB',
  title: 'JobsDB Footer',
  component: Footer,
  sketch,
  initialProps: {
    language: 'en',
    country: 'hk',
    linkRenderer: makeDummyLinkRendererForPath(ROUTE)
  },
  options: [
    {
      label: 'Language',
      type: 'radio',
      states: [
        {
          label: 'English',
          transformProps: props => ({
            ...props,
            language: 'en'
          })
        },
        {
          label: 'Bahasa',
          transformProps: props => ({
            ...props,
            language: 'id'
          })
        },
        {
          label: 'Thai',
          transformProps: props => ({
            ...props,
            language: 'th'
          })
        }
      ]
    },
    {
      label: 'Country',
      type: 'radio',
      states: [
        {
          label: 'Hong Kong',
          transformProps: props => ({
            ...props,
            country: 'hk'
          })
        },
        {
          label: 'Singapore',
          transformProps: props => ({
            ...props,
            country: 'sg'
          })
        },
        {
          label: 'Indonesia',
          transformProps: props => ({
            ...props,
            country: 'id'
          })
        },
        {
          label: 'Thailand',
          transformProps: props => ({
            ...props,
            country: 'th'
          })
        }
      ]
    }
  ]
};
