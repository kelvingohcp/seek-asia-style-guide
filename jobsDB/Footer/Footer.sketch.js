import React from 'react';
import Footer from './Footer';

export const blockSymbols = {
  'JobsDB/Footer/Hong Kong EN': <Footer language="en" country="hk" />,
  'JobsDB/Footer/Hong Kong EN, isExpandedVersion': <Footer language="en" country="hk" isExpandedVersion />,
  'JobsDB/Footer/Indonesia EN': <Footer language="en" country="id" />,
  'JobsDB/Footer/Indonesia ID': <Footer language="id" country="id" />,
  'JobsDB/Footer/Singapore EN': <Footer language="en" country="sg" />,
  'JobsDB/Footer/Thailand EN': <Footer language="en" country="th" />,
  'JobsDB/Footer/Thailand TH': <Footer language="th" country="th" />
};
