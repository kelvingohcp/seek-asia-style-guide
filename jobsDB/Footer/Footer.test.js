import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('Footer component', () => {
  const _Date = Date;

  beforeEach(() => {
    const DATE = new Date('2017');
    global.Date = jest.fn(() => DATE);
    global.Date.getFullYear = _Date.getFullYear;
  });

  afterEach(() => {
    global.Date = _Date;
  });

  describe('Localization', () => {
    it('should render en-hk', () => {
      const wrapper = shallow(<Footer language="en" country="hk" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-id', () => {
      const wrapper = shallow(<Footer language="en" country="id" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-th', () => {
      const wrapper = shallow(<Footer language="en" country="th" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-sg', () => {
      const wrapper = shallow(<Footer language="en" country="sg" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render id-id', () => {
      const wrapper = shallow(<Footer language="id" country="id" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render th-th', () => {
      const wrapper = shallow(<Footer language="th" country="th" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render en-hk fat footer', () => {
      const wrapper = shallow(<Footer language="en" country="hk" isExpandedVersion />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
