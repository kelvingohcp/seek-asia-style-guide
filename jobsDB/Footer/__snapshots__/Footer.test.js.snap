// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Localization should render en-hk 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "Copyright © 1998-{year}, jobsDB.",
            "footer.legalLink": "https://m.jobsdb.com/en-hk/legal.do",
            "footer.legalLinkText": "Legal terms",
            "footer.privacyLink": "https://hk.jobsdb.com/en-hk/pages/terms/privacy-policy",
            "footer.privacyLinkText": "Privacy Statement",
            "footer.rightReserved": "All Rights Reserved.",
            "footer.siteMapLink": "https://hk.jobsdb.com/hk/en/sitemap",
            "footer.siteMapLinkText": "Site Map",
            "footer.termsLink": "https://hk.jobsdb.com/en-hk/pages/terms/terms-conditions",
            "footer.termsLinkText": "Terms & Conditions",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        Copyright © 1998-2017, jobsDB.
        <span
          className="fullversion"
        >
          All Rights Reserved.
        </span>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer component Localization should render en-hk fat footer 1`] = `
<FooterHK
  country="hk"
  domainUrl="jobsdb.com"
  hrefLink={[Function]}
  language="en"
  linkRenderer={[Function]}
/>
`;

exports[`Footer component Localization should render en-id 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "Copyright © 1998-{year}, jobsDB.",
            "footer.legalLink": "https://m.jobsdb.co.id/en-id/legal.do",
            "footer.legalLinkText": "Legal terms",
            "footer.privacyLink": "https://id.jobsdb.com/en-id/pages/terms/privacy-policy",
            "footer.privacyLinkText": "Privacy Statement",
            "footer.rightReserved": "All Rights Reserved.",
            "footer.siteMapLink": "https://id.jobsdb.com/id/en/sitemap",
            "footer.siteMapLinkText": "Site Map",
            "footer.termsLink": "https://id.jobsdb.com/en-id/pages/terms/terms-conditions",
            "footer.termsLinkText": "Terms & Conditions",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        Copyright © 1998-2017, jobsDB.
        <span
          className="fullversion"
        >
          All Rights Reserved.
        </span>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer component Localization should render en-sg 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "Copyright © 1998-{year}, jobsDB.",
            "footer.legalLink": "https://m.jobsdb.com/en-sg/legal.do",
            "footer.legalLinkText": "Legal terms",
            "footer.privacyLink": "https://sg.jobsdb.com/en-sg/pages/terms/privacy-policy",
            "footer.privacyLinkText": "Privacy Statement",
            "footer.rightReserved": "All Rights Reserved.",
            "footer.siteMapLink": "https://sg.jobsdb.com/sg/en/sitemap",
            "footer.siteMapLinkText": "Site Map",
            "footer.termsLink": "https://sg.jobsdb.com/en-sg/pages/terms/terms-conditions",
            "footer.termsLinkText": "Terms & Conditions",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        Copyright © 1998-2017, jobsDB.
        <span
          className="fullversion"
        >
          All Rights Reserved.
        </span>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer component Localization should render en-th 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "Copyright © 1998-{year}, jobsDB.",
            "footer.legalLink": "https://m.jobsdb.co.th/en-th/legal.do",
            "footer.legalLinkText": "Legal terms",
            "footer.privacyLink": "https://th.jobsdb.com/en-th/pages/terms/privacy-policy",
            "footer.privacyLinkText": "Privacy Statement",
            "footer.rightReserved": "All Rights Reserved.",
            "footer.siteMapLink": "https://th.jobsdb.com/th/en/sitemap",
            "footer.siteMapLinkText": "Site Map",
            "footer.termsLink": "https://th.jobsdb.com/en-th/pages/terms/terms-conditions",
            "footer.termsLinkText": "Terms & Conditions",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        Copyright © 1998-2017, jobsDB.
        <span
          className="fullversion"
        >
          All Rights Reserved.
        </span>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer component Localization should render id-id 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "Hak Cipta© 1998-{year}, jobsDB.",
            "footer.legalLink": "https://m.jobsdb.co.id/en-id/legal.do",
            "footer.legalLinkText": "Syarat Hukum",
            "footer.privacyLink": "https://id.jobsdb.com/id-id/pages/terms/privacy-policy",
            "footer.privacyLinkText": "Pernyataan Privasi",
            "footer.rightReserved": "Semua Hak Dilindungi.",
            "footer.siteMapLink": "https://id.jobsdb.com/id/id/sitemap",
            "footer.siteMapLinkText": "Sitemap",
            "footer.termsLink": "https://id.jobsdb.com/id-id/pages/terms/terms-conditions",
            "footer.termsLinkText": "Syarat dan ketentuan",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        Hak Cipta© 1998-2017, jobsDB.
        <span
          className="fullversion"
        >
          Semua Hak Dilindungi.
        </span>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer component Localization should render th-th 1`] = `
<footer
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="pullRight"
    >
      <FooterLinks
        messages={
          Object {
            "footer.copyright": "สงวนลิขสิทธิ์ 1998-{year} jobsDB",
            "footer.legalLink": "https://m.jobsdb.co.th/en-th/legal.do",
            "footer.legalLinkText": "ข้อกำหนดทางกฎหมาย",
            "footer.privacyLink": "https://th.jobsdb.com/th-th/pages/terms/privacy-policy",
            "footer.privacyLinkText": "คำชี้แจงสิทธิส่วนบุคคล",
            "footer.rightReserved": "",
            "footer.siteMapLink": "https://th.jobsdb.com/th/th/sitemap",
            "footer.siteMapLinkText": "แผนที่เว็บไซต์",
            "footer.termsLink": "https://th.jobsdb.com/th-th/pages/terms/terms-conditions",
            "footer.termsLinkText": "ข้อกำหนดและเงื่อนไข",
          }
        }
      />
    </div>
    <div
      className="pullLeft"
    >
      <p
        className="copyright"
      >
        สงวนลิขสิทธิ์ 1998-2017 jobsDB
        <span
          className="fullversion"
        />
      </p>
    </div>
  </div>
</footer>
`;
